draft-irtf-cfrg-hybrid-kems.md

```diff
@@ -162,61 +199,92 @@ The following terms are used throughout this document:
+     | GenerateKeyPair |
+     |        or       |
+     |  DeriveKeyPair  |
+     +--------+--------+
+              |
+    +---------+----------+
+    |                    |
+    V                    V
+
+    ek                  dk
+
+    |                    |
+    |                    |
+    V                    V
++--------+    ct    +--------+
+| Encaps |--------->| Decaps |
++--------+          +--------+
+    |                    |
+    |                    |
+    V                    V
+
+    ss        ==        ss
+~~~
+
+A Key Encapsulation Mechanism (KEMs) comprises the following algorithms:
+
+- `GenerateKeyPair() -> (ek, dk)`: A randomized algorithm that generates a
+  public encapsulation key `ek` and a secret decapsulation key `dk`, each of
+  which are byte strings.
+- `DeriveKeyPair(seed) -> (ek, dk)`: A deterministic algorithm that takes as
+  input a seed `seed` and generates a public encapsulation key `ek` and a secret
+  decapsulation key `dk`, each of which are byte strings.
+- `Encaps(ek) -> (ct, ss)`: A probabilistic encapsulation
+  algorithm, which takes as input a public encapsulation key `ek` and outputs
+  a ciphertext `ct` and shared secret `ss`.
+- `Decaps(dk, ct) -> ss`: A decapsulation algorithm, which takes
+  as input a secret decapsulation key `dk` and ciphertext `ct` and outputs a
+  shared secret `ss`.
+
+A KEM may also provide a deterministic version of `Encaps` (e.g., for purposes
+of testing):
+
+- `EncapsDerand(ek, randomness) -> (ct, shared_secret)`: A deterministic
    encapsulation algorithm, which takes as input a public encapsulation key
-   `pk` and randomness `randomness`, and outputs a ciphertext `ct` and shared
+   `ek` and randomness `randomness`, and outputs a ciphertext `ct` and shared
    secret `shared_secret`.
 
-Finally, KEMs are also parameterized with the following constants:
+We assume that the values produced and consumed by the above functions are all
+byte strings, with fixed lengths:
+
+- `Nseed`: The length of a key seed (input to DeriveKeyPair)
+- `Nek`: The length of a public encapsulation key
+- `Ndk`: The length of a secret decapsulation key
+- `Nct`: The length of a ciphertext produced by Encaps
+- `Nss`: The length of a shared secret produced by Encaps or Decaps
 
-- Nseed, which denotes the number of bytes for a key seed;
-- Npk, which denotes the number of bytes in a public encapsulation key;
-- Nsk, which denotes the number of bytes in a private decapsulation key; and
-- Nct, which denotes the number of bytes in a ciphertext.
+This interface is effectively the same as the one defined in the Hybrid Public
+Key Encryption (HPKE) specification {{?RFC9180}}.  The only difference is that
+here we assume that all values are byte strings, whereas in HPKE keys are opaque
+by default and serialized or deserialized as needed.  We also use slightly
+different terminology for keys, emphasizing "encapsulation" and "decapsulation"
+as opposed to "public" and "secret".
```

Comment:
True, but why mention it?

Reply:
In case people have already read HPKE and are looking at this.  I have added markup to make it an inset note / aside.  I'm not that committed to it; happy to delete it if people feel strongly.